### PR TITLE
Add a pure Wasm image classification benchmark

### PR DESCRIPTION
**Pull request by adambratschikaye**
Add a benchmark which performs image classification in pure Wasm using tract and ONNX. This can be useful to track the performance of float heavy computations.

### CODE REVIEW

**Comment by adambratschikaye** [comment]
I think the rebuild failure is due to libsodium now requiring zig 0.12, but the Dockerfile uses zig 0.10.1.

**Comment by adambratschikaye** [comment]
Rebuild failure should be fixed in https://github.com/bytecodealliance/sightglass/pull/271

File: benchmarks/tract-onnx-image-classification/setup.sh [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+set -euxo pipefail
+
+MODEL_URL="https://www.github.com/onnx/models/raw/main/validated/vision/classification/mobilenet/model/mobilenetv2-7.onnx?download=:"
+MODEL_FILE="assets/mobilenetv2-7.onnx"
+
+if [ ! -f $MODEL_FILE ]; then
+	echo "Downloading model to $MODEL_FILE"
+
+	if which wget >/dev/null ; then
+	wget $MODEL_URL -O $MODEL_FILE
```
Comment (jlb6740) [inline benchmarks/tract-onnx-image-classification/setup.sh:12]: An assets directory needs to be created first.

**Comment by jlb6740** [comment]
@adambratschikaye I was not familiar with tract but was introduced by your issue here: https://github.com/bytecodealliance/wasmtime/pull/8313, so thanks for submitting. I think getting this in now is great but ultimately maybe this directory could be refactored to include multiple inferencing scenarios.

File: benchmarks/tract-onnx-image-classification/setup.sh (hunk 1 above)
Comment (adambratschikaye) [reply benchmarks/tract-onnx-image-classification/setup.sh:12]: Oops, thanks I'll fix that.

**Comment by jlb6740** [comment]
@adambratschikaye Looks like you were right. Not sure the corruption, but it was using an incorrect cached version of the wasmtime-engine. I removed the cached image for macos  and all tests past now.

**Review by jlb6740 — APPROVED** [review]

**Pull request merged**